lib/ramen/test.js

```diff
@@ -0,0 +1,152 @@
+        analytics.page();
+        analytics.assert(!window.ramenSettings);
+        analytics.assert(!window.Ramen);
+      });
+
+      it('should call #load', function(){
+        analytics.initialize();
+        analytics.page();
+        analytics.called(ramen.load);
+      });
+    });
+  });
+
+  describe('loading', function(){
+    it('should load', function(done){
+      analytics.load(ramen, done);
+    });
+  });
+
+  describe('after loading', function(){
+    beforeEach(function(done){
+      analytics.once('ready', done);
+      analytics.initialize();
+    });
+
+    describe('#identify', function(){
+      beforeEach(function(){
+        analytics.stub(window.Ramen, 'go');
+      });
+
+      it('should not call Ramen.go if only id is passed', function(){
+        analytics.identify('id');
+        analytics.assert(!window.ramenSettings);
+        analytics.didNotCall(window.Ramen.go);
+
+        analytics.identify('id');
+        analytics.assert(!window.ramenSettings);
+        analytics.didNotCall(window.Ramen.go);
+      });
+
+      it('should call Ramen.go and set correct attributes if just email passed', function(){
+        var email = 'email@example.com';
+        analytics.identify('id', { email: email });
+        analytics.assert(window.ramenSettings.organization_id === '6389149');
+        analytics.assert(window.ramenSettings.user.id === 'id');
+        analytics.assert(window.ramenSettings.user.name === email);
+        analytics.assert(window.ramenSettings.user.email === email);
+        analytics.called(window.Ramen.go);
+      });
+
+      it('should call Ramen.go and set correct attributes if email & name passed', function(){
+        var email = 'email@example.com';
+        var name = 'ryan+segment@ramen.is';
+        analytics.identify('id', { email: email, name: name });
+        analytics.assert(window.ramenSettings.organization_id === '6389149');
+        analytics.assert(window.ramenSettings.user.id === 'id');
+        analytics.assert(window.ramenSettings.user.name === name);
+        analytics.assert(window.ramenSettings.user.email === email);
+        analytics.called(window.Ramen.go);
+      });
+
+      it('should pass along integration options', function(){
+        var email = 'email@example.com';
+        var name = 'ryan+segment@ramen.is';
+        var auth_hash = 'authy_hasy';
+        var timestamp = new Date() / 1000;
+        var custom_links = [{href: 'https://ramen.is/support', title: 'Hello'}];
+        var labels = ['use', 'ramen!'];
+        var environment = 'staging';
+        var logged_in_url = 'https://align.ramen.is/manage';
+        var unknown_future_opt = '11';
+        var unknown_future_user_opt = 'user 11';
+
+        analytics.identify('id', {email: email, name: name},
+          {
+            integrations: {
+              Ramen: {
+                unknown_future_opt: unknown_future_opt,
+                environment: environment,
+                timestamp: timestamp,
```

Comment:
Actually as I went to do this, I realized that we need to let them pass in `timestamp` here since it factors into the `auth_hash` and the timestamp when analytics.identify() fires will not be the same as the time the `auth_hash` is generated server-side.